packages/plugins/typescript/apollo-client-helpers/src/index.ts

```diff
@@ -0,0 +1,84 @@
+import { Types, PluginValidateFn, PluginFunction } from '@graphql-codegen/plugin-helpers';
+import { GraphQLSchema, isInterfaceType, isObjectType } from 'graphql';
+import { extname } from 'path';
+import { ApolloClientHelpersConfig } from './config';
+
+export const plugin: PluginFunction<ApolloClientHelpersConfig> = (
+  schema: GraphQLSchema,
+  documents: Types.DocumentFile[],
+  config: ApolloClientHelpersConfig
+) => {
+  const results: Types.ComplexPluginOutput[] = [];
+
+  results.push(generateTypePoliciesSignature(schema, config));
+
+  return {
+    prepend: results.reduce((prev, r) => [...prev, ...(r.prepend || [])], []),
+    append: results.reduce((prev, r) => [...prev, ...(r.append || [])], []),
+    content: results.map(r => r.content).join('\n'),
+  };
+};
+
+function generateTypePoliciesSignature(
+  schema: GraphQLSchema,
+  config: ApolloClientHelpersConfig
+): Types.ComplexPluginOutput {
+  const typeMap = schema.getTypeMap();
+  const perTypePolicies: string[] = [];
+  const typedTypePolicies = Object.keys(typeMap).reduce((prev, typeName) => {
+    const type = typeMap[typeName];
+
+    if (!typeName.startsWith('__') && (isObjectType(type) || isInterfaceType(type))) {
+      const fieldsNames = Object.keys(type.getFields()).filter(f => !f.startsWith('__'));
```

Comment:
it is allowed to specify nested keys (see the `Book` example using `["title", "author", ["name"]]`) here: https://www.apollographql.com/docs/react/caching/cache-configuration/#generating-unique-identifiers. Unless I'm doing something weird, that produces a type error here

Reply:
Are you sure it's an actual issue? I can see that it works perfectly: 
![image](https://user-images.githubusercontent.com/3680083/97154860-2429c480-177d-11eb-8f6c-dd5458cef78d.png)

Which TS version are you using